File: src/main/java/eu/happybe/openapi/mysql/query/AddTokensQuery.java
package eu.happybe.openapi.mysql.query;

import eu.happybe.openapi.mysql.AsyncQuery;

import java.sql.SQLException;
import java.sql.Statement;

public class AddTokensQuery extends AsyncQuery {

    public String player;
    public int coins;

    public AddTokensQuery(String player, int coins) {
        this.player = player;
        this.coins = coins;
    }

    @Override
    public void query(Statement statement) throws SQLException {
        if(this.coins < 0) {
            statement.executeUpdate("UPDATE HB_Values SET Tokens=Tokens-" + Math.abs(this.coins) + " WHERE Name='"+this.player+"'");
            return;
        }
        statement.executeUpdate("UPDATE HB_Values SET Tokens=Tokens+" + coins + " WHERE Name='"+this.player+"';");
    }
}

File: src/main/java/eu/happybe/openapi/mysql/query/AddTokensQuery.kt
package eu.happybe.openapi.mysql.query

import eu.happybe.openapi.mysql.AsyncQuery
import java.sql.SQLException
import java.sql.Statement

class AddTokensQuery(var player: String, var coins: Int) : AsyncQuery() {
    @Throws(SQLException::class)
    override fun query(statement: Statement) {
        if (coins < 0) {
            statement.executeUpdate("UPDATE HB_Values SET Tokens=Tokens-" + Math.abs(coins) + " WHERE Name='" + player + "'")
            return
        }
        statement.executeUpdate("UPDATE HB_Values SET Tokens=Tokens+" + coins + " WHERE Name='" + player + "';")
    }
}